### PR TITLE
Use Cmake to generate files from codec.txt automatically on AArch64.

### PR DESCRIPTION
**Pull request by fhahn**
This patch removes
 * core/arch/aarch64/decode_gen.h,
 * core/arch/aarch64/encode_gen.h
 * core/arch/aarch64/opcode.h
 * core/arch/aarch64/opcode_names.h

and instead adds Cmake rules to automatically generate them from
codec.txt.

The main advantage is that this dramatically reduces the diff for codec
changes on AArch64, as it will not contain tons of changes in the
auto-generated files.

Change-Id: I88c38835c887d252fe0fa4e406879e3b5d682c19

### CODE REVIEW

File: core/lib/genapi.pl [hunk 1]
```diff
@@ -212,6 +212,11 @@ sub copy_file
     push(@headers, "$core/annotations.h");
 }
 
+# AArch64's opcode.h is auto-generated.
+if (defined($defines{"AARCH64"})) {
+    push(@headers, "$dir/../opcode.h");
```
Comment (derekbruening) [inline core/lib/genapi.pl]: So this assumes the output dir is one deep in the build dir.  Worth a comment about the assumption.

File: make/CMake_aarch64_gen_codec.cmake [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+# DAMAGE.
+
+# Commands to automatically create the codec files from core/arch/aarch64/codec.txt
+find_package(PythonInterp)
+
+if (NOT PYTHONINTERP_FOUND)
+message(FATAL_ERROR "Python interpreter not found")
```
Comment (derekbruening) [inline make/CMake_aarch64_gen_codec.cmake]: Indentation off

File: make/CMake_aarch64_gen_codec.cmake [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+set(AARCH64_CODEC_GEN_SRCS
+  ${PROJECT_BINARY_DIR}/opcode.h
+  ${PROJECT_BINARY_DIR}/decode_gen.h
+  ${PROJECT_BINARY_DIR}/encode_gen.h
+  ${PROJECT_BINARY_DIR}/opcode_names.h
+)
+set_source_files_properties(${AARCH64_CODEC_GEN_SRCS} PROPERTIES GENERATED true)
+# Auto-generate decoder files from codec.txt.
+add_custom_command(
+OUTPUT  ${AARCH64_CODEC_GEN_SRCS}
```
Comment (derekbruening) [inline make/CMake_aarch64_gen_codec.cmake]: All other instances indent these args

File: make/CMake_aarch64_gen_codec.cmake (hunk 2 above)
Comment (fhahn) [reply make/CMake_aarch64_gen_codec.cmake]: Sorry about that, I somehow messed up the indentation in this file

**Pull request merged**